src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/PropertiesFileTransformer.kt

```diff
@@ -122,8 +123,10 @@ public open class PropertiesFileTransformer @Inject constructor(
   @get:Input
   public open val charsetName: Property<String> = objectFactory.property(Charsets.ISO_8859_1.name())
 
-  @get:Internal
-  public open var keyTransformer: (String) -> String = IDENTITY
+  @Transient // Input property should be Serializable, but we don't want to serialize the lambda.
+  @get:Optional
+  @get:Input
+  public open var keyTransformer: ((String) -> String)? = null
```

Comment:
It can't work correctly with caching, see https://github.com/GradleUp/shadow/pull/1208#discussion_r1931767699.

Reply:
Reverts this in d39b7de3abfa0c9a3c618eab9bfae50a064254d4.